### ISSUE TITLE
CCM doesn't fill MetalLB configmap password string when password is set for project BGP settings.

### ISSUE THREAD

**Issue by enkelprifti98** [equinix/cloud-provider-equinix-metal#161]
The CCM leaves the password string empty in the MetalLB configmap when an Equinix Metal user has configured their project BGP settings with an MD5 password.

To replicate this issue:

1. Deploy a kubernetes cluster with kubelet set to cloudprovider=external.
2. Enable Local BGP for the project the cluster is in and set a BGP password.
3. Deploy CCM with the following environment variable so it automates MetallLB configmap:
```
        env:
          - name: METAL_LB
            value: "metallb://"
```
4. Deploy MetalLB through the official instructions: 
https://metallb.universe.tf/installation/#installation-by-manifest
5. Create a dummy metallb configmap with the following content:
```
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: metallb-system
  name: config
data:
  config: |
```

After deploying the metallb configmap, the CCM should have filled out the BGP peers and any elastic IP addresses. When the user has set a BGP password for their project, the metalLB will have an empty string for the password field under each BGP peer:

`password: ""`

Since the password string is empty, metallb cannot peer with the Equinix Metal router as it requires a BGP password to establish the session.

**Comment by deitch** [comment]
I can confirm this is an issue. We need to put it as annotations as well as in the metallb configmap.

**Comment by jeff-anderson-dfinity** [comment]
> I can confirm this is an issue. We need to put it as annotations as well as in the metallb configmap.

Is there a current annotation and configmap key that can be used?

**Comment by deitch** [comment]
Not really. The annotation isn't in place yet, and the configmap gets overwritten (which is the heart of the issue). We are in active discussions about the right way to do this.

**Comment by enkelprifti98** [comment]
This issue was addressed with PR #163 

After testing CCM v3.1.0, the MetalLB configmap gets the BGP password.
